pymedphys/tests/dicom/test_anonymise.py

```diff
@@ -169,6 +169,27 @@ def test_anonymised_dataset_with_empty_patient_sex():
     assert hardcode_replace_ds["PatientSex"].value is None
 
 
+@pytest.mark.slow
+@pytest.mark.pydicom
+def test_alternative_identifying_keywords():
+    default_keyword_list = list(IDENTIFYING_KEYWORDS)
+    alternative_keyword_list = list(default_keyword_list)
+    alternative_keyword_list.append("SOPInstanceUID")
+    test_file_path = get_treatmentrecord_test_file_path()
+    ds_test = pydicom.dcmread(test_file_path, force=True)
+    anon_private_filepath = anonymise_file(
+        test_file_path,
+        delete_private_tags=True,
+        alternative_identifying_keywords=alternative_keyword_list,
+    )
```

Comment:
Sometimes these tests need to be able to be run in parallel. Therefore creating and deleting files amongst the test data can cause really weird errors down the line.

Can you make use of the tempfile library when creating files. Here's an example of what I mean:

https://github.com/pymedphys/pymedphys/blob/7959e19b256a3befb965817741002a3e6c6235d9/pymedphys/tests/trf/test_decode.py#L84-L89

When you use the tempfile library you won't actually have to delete the file at the end either.

Reply:
OK.  But note that what I deleted was the newly created anonymised file whose SOP Instance UID was changed.

Reply:
Yup, the weird thing that can happen is, let's say this is being tested on the same machine for all of Python 3.6, 3.7, and 3.8. What can happen here is the Python 3.6 version might create the file, and then maybe the 3.7 version errors with a `FileExistsError`, or maybe it ignores that, uses the version that was created with Python 3.6, but then at the last moment while it's reading maybe Python 3.6 tries to delete it, and gets file lock/user access errors. All sorts of pain. And intermittent.

Reply:
Understood and implemented.